pkgs/tools/misc/calamares/waylandkbd.patch

```diff
@@ -0,0 +1,25 @@
+diff --git a/src/modules/keyboard/Config.cpp b/src/modules/keyboard/Config.cpp
+index 720588810..d4f4e4696 100644
+--- a/src/modules/keyboard/Config.cpp
++++ b/src/modules/keyboard/Config.cpp
+@@ -219,7 +219,10 @@ Config::xkbApply()
+                                                { m_additionalLayoutInfo.additionalVariant, m_selectedVariant },
+                                                m_additionalLayoutInfo.groupSwitcher ) );
+ 
+-
++        QProcess::execute( "bash", { "-c", "if command -v gsettings; then gsettings set org.gnome.desktop.input-sources sources \"[('xkb','"+
++                            m_selectedLayout + ((!m_selectedVariant.isEmpty()) ? "+"+m_selectedVariant : "") +
++                            "'),('xkb','"+m_additionalLayoutInfo.additionalLayout+ ((!m_additionalLayoutInfo.additionalLayout.isEmpty()) ? "+"+m_additionalLayoutInfo.additionalVariant : "") +
++                            "')]\"; fi" });
+         cDebug() << "xkbmap selection changed to: " << m_selectedLayout << '-' << m_selectedVariant << "(added "
+                  << m_additionalLayoutInfo.additionalLayout << "-" << m_additionalLayoutInfo.additionalVariant
+                  << " since current layout is not ASCII-capable)";
+@@ -227,6 +230,8 @@ Config::xkbApply()
+     else
+     {
+         QProcess::execute( "setxkbmap", xkbmap_layout_args( m_selectedLayout, m_selectedVariant ) );
++        QProcess::execute( "bash", { "-c", "if command -v gsettings; then gsettings set org.gnome.desktop.input-sources sources \"[('xkb','" + 
```

Comment:
What does value returned by `xkbmap_layout_args` looks like? I might be better to pass this in as the first argument to bash and then use `$1` in bash to accidentally or mal-intended code injection.

Reply:
So What I mean is that you could build the whole string in c++: 
`[('xkb','<yourvariant>')]` and then bass this in as a seperate argument to bash.

Reply:
The alternative would be to just run `QProcess::execute` opportunistically on `gsettings` and catch ENOENT errors, than you wouldn't need bash at all.

Reply:
```
$ sh -c 'echo $0 = $1' foo bar
foo = bar
```

Reply:
Was using `$1` instead of `$0` 🤦🏽